### DISCUSSION TITLE
Icons on server without internet acces.

### DISCUSSION THREAD

**Discussion by vdovhanych** [dani-garcia/vaultwarden#1606] [Q&A] [answered]
Hi, i was wondering if there is a way for the server to load icons without having acces to the internet ?

- Bitwarden local icons server ?
- Preload all the icons in cahce folder ?

Is it even possible to download all the icons from bitwarden website to put in in the cache folder ? 

Main reason for this is i dont want to have bitwarden open to itnertnet and also i don't want to send out information on what services i have some sort of accout. Thanks for any help guys.

**Comment by BlackDex** [comment, marked as answer]
I'm not sure how you want to download icon's without internet access at all.
You have to download them via the internet in the end.
Also, those sites already know that you have an account there probably ;).

All icon's are downloaded by the bitwarden_rs application it self, there is no usage of external services to get the icon's.

If you really want you could download the icons your self, and name them domain.tld.png.
Also, make sure you set the icon_cache_ttl to 0, and disable icon downloading, that should keep the icons as you have them.

**Reply by vdovhanych** [reply]
Hi, thanks for the quick response. Yes, I know I have to download them in the end. My thinking is that if you download all the icons at once there is no information as to what types of services you have accounts on. From what I've seen the bitwrdent_rs sends a request to the official bitwarden domain (eg. [https://icons.bitwarden.net/google.com/icon.png](https://icons.bitwarden.net/google.com/icon.png))  and thus this type of information is sent out in the normal usage of bitwarden when it has internet access and you enable icons. In this case, bitwarden knows I have a login to that and that domain so it's not necessarily bad but that information is out there and I would rather keep it hidden even from bitwarden.

**Reply by BlackDex** [reply]
I don't know where you have seen that. I build the icon downloading my self in bitwarden_rs, it doesn't use that service at all.
It did in the very beginning, but not for a long time now, so if that was your only concern, then i would just enable the icon downloading, set a high TTL or even 0, and be done with it :).

**Reply by vdovhanych** [reply]
Oh ok, i didn't do much digging in the code but i have seen that in official bitwarden docs and assumed the bitwarden_rs uses same implementation. So this means the server will grab favicon from the website itself ?

**Reply by BlackDex** [reply]
the bitwarden_rs application will fetch the icons it self:
https://github.com/dani-garcia/bitwarden_rs/blob/master/src/api/icons.rs

No usage of any third-party services at all.

**Reply by vdovhanych** [reply]
Amazing, thanks for the clarification and the work you've done on this.
